services/permits/app/permit_conditions/resources/permit_condition_resource.py

```diff
@@ -42,56 +50,86 @@ async def extract_permit_conditions(file: UploadFile = File(...)) -> PermitCondi
 
     # Write the uploaded file to a temporary file
     # so it can be processed by the pipeline.
-    tmp = store_temporary(file)
+    tmp = store_temporary(file, suffix=".pdf")
 
     try:
-        pipeline = permit_condition_pipeline()
-
-        return pipeline.run(
-            {
-                "pdf_converter": {"file_path": tmp.name},
-                "prompt_builder": {
-                    "template_variables": {
-                        "max_pages": 6,
-                    }
-                },
-            }
-        )["validator"]
-    finally:
+        res = run_permit_condition_pipeline.delay(tmp.name, {"original_file_name": file.filename, 'size': file.size, 'content_type': file.content_type})
+        return JobStatus(id=res.task_id, status=res.status, meta=res.info)
+    except:
         tmp.close()
+        raise
+
+@router.get("/permit_conditions/status")
+def status(task_id: str) -> JobStatus:
+    """
+    Get the status of a permit conditions extraction job.
+    Args:
+        The task ID of the job.
+    """
+    res = run_permit_condition_pipeline.app.AsyncResult(task_id)
+    return JobStatus(id=res.task_id, status=res.status, meta=res.info)
+
+@router.get("/permit_conditions/results")
+def results(task_id: str) -> PermitConditions:
+    """
+    Get the results of a permit conditions extraction job.
+    Args:
+        The task ID of the job.
+    Raises:
+        400 Bad Request: If the task has not completed successfully
+    """
+    res = run_permit_condition_pipeline.app.AsyncResult(task_id)
 
+    if res.status == "SUCCESS":
+        return res.get()
+    elif res.status == "FAILURE":
+        raise HTTPException(500, detail=f"Task failed to complete: {res.status}")
+    else:
+        raise HTTPException(202, detail=f"Task has not completed yet. Current status: {res.status}")
```

Comment:
Does it work to return a 200-type code with HTTPException? If it works, cool, but it looks kinda fishy to me. 😄 

May want to consider an enum with SUCCESS/FAILURE/PROGRESS statuses.

Reply:
It does indeed work, but yup not the most clear.